### PR TITLE
SJ - PARSER DATA SOURCE DROP-DOWN FORM BUG

### PR DESCRIPTION
**Pull request by Reinkaos**
**Acceptance Criteria**
- "change data source" button should show data source pre-selected on current data source
https://manager.digitalnz.org/parsers/5dc135ac6f7fe178ce949c62/edit

**Notes**
- Currently the data source drop-down is pre populated with the top option.
It should instead show the current data source (which in the above link should be "atl")

### CODE REVIEW

File: app/views/parsers/_parser.html.erb [hunk 1]
```diff
@@ -162,7 +162,7 @@
   <%= custom_form_with model: @parser, local: true do |f| %>
     <%= f.label :partner, 'Contributor' %>
     <%= f.select :partner, Partner.all.sort(name: 1).map { |p| [p.name, p.name] }, selected: @parser.partner.name %>
-    <%= f.label :source, 'Data source' %>
+    <%= f.label :source, 'Data source', data: { 'current_source_id': @parser.source_id } %>
```
Comment (paul-mesnilgrente) [inline app/views/parsers/_parser.html.erb:165]: I think you can avoid this javascript by setting the `value` on the `grouped_collection_select`. No?

Comment (Reinkaos) [reply app/views/parsers/_parser.html.erb:165]: na it's kind of strange but the value is currently set, but when the javascript that appends and removes options runs we lose the value. Even though the tag still has selected on it

Comment (Reinkaos) [reply app/views/parsers/_parser.html.erb:165]: if you don't run this JS at all, the form works correctly it just shows all of the options

Comment (paul-mesnilgrente) [reply app/views/parsers/_parser.html.erb:165]: Ah yeah, I thought it was on the first display. All good then :)

Comment (Reinkaos) [reply app/views/parsers/_parser.html.erb:165]: 😄

**Pull request merged**